### PR TITLE
Add more public APIs to crash test

### PR DESCRIPTION
**Pull request by hx235**
**Context/Summary:** 
As titled. Bonus: found that PromoteL0 called with other concurrent PromoteL0 will return non-okay error so clarify the API.

**Test:**
CI

### CODE REVIEW

File: db_stress_tool/db_stress_test_base.cc [hunk 1]
```diff
@@ -2428,6 +2458,29 @@ void StressTest::TestCompactFiles(ThreadState* thread,
   }
 }
 
+void StressTest::TestPromoteL0(ThreadState* thread,
+                               ColumnFamilyHandle* column_family) {
+  int target_level = thread->rand.Next() % options_.num_levels;
+  Status s = db_->PromoteL0(column_family, target_level);
```
Comment (ajkr) [inline db_stress_tool/db_stress_test_base.cc:2464]: Never heard of this API. Hope it works.

**Review by ajkr — APPROVED** [review]

File: db_stress_tool/db_stress_test_base.cc [hunk 2]
```diff
@@ -2358,6 +2375,19 @@ Status StressTest::TestGetPropertiesOfAllTables() const {
   return db_->GetPropertiesOfAllTables(&props);
 }
 
+void StressTest::TestGetUpdatesSince(ThreadState* thread) {
+  SequenceNumber latest_seqno = db_->GetLatestSequenceNumber();
+  SequenceNumber seq = thread->rand.Next64() % latest_seqno + 1;
+  std::unique_ptr<TransactionLogIterator> iter;
+  Status s = db_->GetUpdatesSince(seq, &iter);
```
Comment (ajkr) [inline db_stress_tool/db_stress_test_base.cc:2382]: Doesn't the user have to set some option for this to return results that make sense?

File: db_stress_tool/db_stress_test_base.cc (hunk 1 above)
Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc:2464]: There is still internal use case though not big. I found this API useful to move files trivially to non-L0 in UT https://github.com/facebook/rocksdb/pull/12628/files#diff-68a832436b821ea3d361ae2ab06d28f663569e352a4ca37cadbc30a0f7bd23c3R7630 ...

File: db_stress_tool/db_stress_test_base.cc (hunk 2 above)
Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc:2382]: Yeah - the wal_xxx parameters ned to be set very high. Still it was a bit flaky in my local run. Let me try again and add `TestGetUpdatesSince` in another PR for easier testing.
